testsupport/tiers/tier_setup.go

```diff
@@ -0,0 +1,189 @@
+	SpaceRolesTier *toolchainv1alpha1.NSTemplateTier
+}
+
+type CustomNSTemplateTierModifier func(*HostAwaitility, *CustomNSTemplateTier) error
+
+func WithClusterResources(otherTier *toolchainv1alpha1.NSTemplateTier) CustomNSTemplateTierModifier {
+	return func(hostAwait *HostAwaitility, tier *CustomNSTemplateTier) error {
+		tier.ClusterResourcesTier = otherTier
+		// configure the "wrapped" NSTemplateTier
+		tmplRef, err := duplicateTierTemplate(hostAwait, otherTier.Namespace, tier.Name, otherTier.Spec.ClusterResources.TemplateRef)
+		if err != nil {
+			return err
+		}
+		tier.Spec.ClusterResources = &toolchainv1alpha1.NSTemplateTierClusterResources{
+			TemplateRef: tmplRef,
+		}
+		return nil
+	}
+}
+
+func WithNamespaceResources(otherTier *toolchainv1alpha1.NSTemplateTier) CustomNSTemplateTierModifier {
+	return func(hostAwait *HostAwaitility, tier *CustomNSTemplateTier) error {
+		tier.NamespaceResourcesTier = otherTier
+		// configure the "wrapped" NSTemplateTier
+		tier.Spec.Namespaces = make([]toolchainv1alpha1.NSTemplateTierNamespace, len(otherTier.Spec.Namespaces))
+		for i, def := range otherTier.Spec.Namespaces {
+			tmplRef, err := duplicateTierTemplate(hostAwait, otherTier.Namespace, tier.Name, def.TemplateRef)
+			if err != nil {
+				return err
+			}
+			tier.Spec.Namespaces[i].TemplateRef = tmplRef
+		}
+		return nil
+	}
+}
+
+func WithSpaceRoles(otherTier *toolchainv1alpha1.NSTemplateTier) CustomNSTemplateTierModifier {
+	return func(hostAwait *HostAwaitility, tier *CustomNSTemplateTier) error {
+		tier.SpaceRolesTier = otherTier
+		// configure the "wrapped" NSTemplateTier
+		tier.Spec.SpaceRoles = make(map[string]toolchainv1alpha1.NSTemplateTierSpaceRole, len(otherTier.Spec.SpaceRoles))
+		for name, def := range otherTier.Spec.SpaceRoles {
+			tmplRef, err := duplicateTierTemplate(hostAwait, otherTier.Namespace, tier.Name, def.TemplateRef)
+			if err != nil {
+				return err
+			}
+			tier.Spec.SpaceRoles[name] = toolchainv1alpha1.NSTemplateTierSpaceRole{
+				TemplateRef: tmplRef,
+			}
+		}
+		return nil
+	}
+}
+
+func CreateCustomNSTemplateTier(t *testing.T, hostAwait *HostAwaitility, name string, baseTier *toolchainv1alpha1.NSTemplateTier, modifiers ...CustomNSTemplateTierModifier) *CustomNSTemplateTier {
+	tier := &CustomNSTemplateTier{
+		NSTemplateTier: &toolchainv1alpha1.NSTemplateTier{
+			ObjectMeta: metav1.ObjectMeta{
+				Namespace: baseTier.Namespace,
+				Name:      name,
+				Labels:    map[string]string{"producer": "toolchain-e2e"},
+			},
+			Spec: toolchainv1alpha1.NSTemplateTierSpec{
+				// default values
+				DeactivationTimeoutDays: baseTier.Spec.DeactivationTimeoutDays,
+			},
+		},
+	}
+	// default values
+	for _, modify := range []CustomNSTemplateTierModifier{
+		WithClusterResources(baseTier),
+		WithNamespaceResources(baseTier),
+		WithSpaceRoles(baseTier),
+	} {
+		err := modify(hostAwait, tier)
+		require.NoError(t, err)
+	}
```

Comment:
it could be simplified with:
```suggestion
	modifiers = append([]CustomNSTemplateTierModifier{
		WithClusterResources(baseTier),
		WithNamespaceResources(baseTier),
		WithSpaceRoles(baseTier),
	}, modifiers...)
```
but it's a question if we need the modifiers with the custom values at all since they are not used anywhere yet.

Reply:
but we can also remove these modifiers, since they aren't used for now, indeed...